x/evm/precompiles/cosmos/contract.go

```diff
@@ -253,6 +253,11 @@ func (e *CosmosPrecompile) ExtendedRun(caller vm.ContractRef, input []byte, supp
 			return nil, ctx.GasMeter().GasConsumedToLimit(), types.ErrNonReadOnlyMethod.Wrap(method.Name)
 		}
 
+		// check if execute cosmos is disabled
+		if e.stateDB.EVM().GetDisallowCosmosDispatch() {
+			return nil, ctx.GasMeter().GasConsumedToLimit(), types.ErrExecuteCosmosDisabled.Wrap(method.Name)
```

Comment:
how about return error before charging gas?

```suggestion
if e.stateDB.EVM().GetDisallowCosmosDispatch() {
	return nil, ctx.GasMeter().GasConsumedToLimit(), types.ErrExecuteCosmosDisabled.Wrap(method.Name)
}
ctx.GasMeter().ConsumeGas(EXECUTE_COSMOS_GAS, "execute_cosmos")
if readOnly {
	return nil, ctx.GasMeter().GasConsumedToLimit(), types.ErrNonReadOnlyMethod.Wrap(method.Name)
}

```